src/Component/FormField/UserPermissionGrid/UserPermissionGrid.tsx

```diff
@@ -0,0 +1,268 @@
+      <div
+        className="column-search"
+      >
+        <Input
+          ref={searchInput}
+          placeholder={t('UserPermissionGrid.filterInputPlaceholder')}
+          value={selectedKeys[0]}
+          onChange={e => setSelectedKeys(e.target.value ? [e.target.value] : [])}
+          onPressEnter={() => confirm()}
+        />
+        <Space>
+          <Button
+            type="primary"
+            onClick={() => confirm()}
+            icon={<SearchOutlined />}
+            size="small"
+          >
+            {t('UserPermissionGrid.filterSearchButtonText')}
+          </Button>
+          <Button
+            onClick={() => {
+              clearFilters();
+              confirm();
+            }}
+            size="small"
+          >
+            {t('UserPermissionGrid.filterResetButtonText')}
+          </Button>
+        </Space>
+      </div>
+    ),
+    filterIcon: (filtered: boolean) => (
+      <SearchOutlined
+        style={{
+          color: filtered ? '#1890ff' : undefined
+        }}
+      />
+    ),
+    onFilter: (value, record) =>
+      record[dataIndex]
+        .toString()
+        .toLowerCase()
+        .includes((value as string).toLowerCase()),
+    onFilterDropdownVisibleChange: visible => {
+      if (visible) {
+        setTimeout(() => searchInput.current?.select(), 100);
+      }
+    }
+  });
+
+  const onPermissionSelect = async (permission: PermissionCollectionType, record: DataType) => {
+    setPermissionsLoading(true);
+
+    try {
+      await client[entityType]().setUserInstancePermission(entityId, record.user?.id, permission);
+
+      let dataClone = _cloneDeep(data);
+      let match = dataClone.find(entry => entry.user?.id === record.user?.id);
+      match.permission = permission;
+
+      setData(dataClone);
+    } catch (error) {
+      message.error(t('UserPermissionGrid.updateErrorMsg'));
+      Logger.error(error);
+    } finally {
+      setPermissionsLoading(false);
+    }
+  };
+
+  const onDeleteClick = async (record: DataType) => {
+    setPermissionsLoading(true);
+
+    try {
+      await client[entityType]().deleteUserInstancePermission(entityId, record.user?.id);
+
+      let dataClone = data.filter(entry => entry.user?.id !== record.user?.id);
+
+      setData(dataClone);
+    } catch (error) {
+      message.error(t('UserPermissionGrid.deleteErrorMsg'));
+      Logger.error(error);
+    } finally {
+      setPermissionsLoading(false);
+    }
+  };
+
+  const onSave = async () => {
+    setPermissionsLoading(true);
+
+    try {
+      setPermissions(await client[entityType]().getUserInstancePermissions(entityId));
+    } catch (error) {
+      message.error(t('UserPermissionGrid.loadErrorMsg'));
+      Logger.error(error);
+    } finally {
+      setPermissionsLoading(false);
+    }
+  };
+
+  const columns: ColumnsType<DataType> = [{
+    title: t('UserPermissionGrid.userColumnTitle'),
+    dataIndex: 'name',
+    key: 'name',
+    defaultSortOrder: 'descend',
```

Comment:
since the sorting is by user name, shouldn't it be ascending, instead of descending?

Reply:
Good point, fixed in 5c30c0b2305b483e9b88a0719c8c9d0d13ad3fca.